File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/1293.rb
while linha = gets
    ls = linha.split(' ')
        xa = (ls[0]).to_f
        ya = (ls[1]).to_f
        xb = (ls[2]).to_f
        yb = (ls[3]).to_f
        cmd = (ls[4]).to_f
        enf = (ls[5]).to_f
        
        ab = Math.sqrt((xa - xb)*(xa - xb) + (ya - yb)*(ya - yb))
        
        cmd = (cmd * 3.14159265358979323846264338327950288419716939937510) / 180
        enf = (enf * 3.14159265358979323846264338327950288419716939937510) / 180
        
        puts "%0.03f" % (ab * (1/Math.tan(cmd) + 1/Math.tan(enf)))
end

File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/1293.lua
function split(inputstr, sep)
    if sep == nil then
            sep = "%s"
    end
    local t={}
    for str in string.gmatch(inputstr, "([^"..sep.."]+)") do
            table.insert(t, str)
    end
    return t
end

while true do
    linha = io.read()
    if linha == nil then break end
    ls = split(linha)

    xa = tonumber(ls[1])
    ya = tonumber(ls[2])
    xb = tonumber(ls[3])
    yb = tonumber(ls[4])
    cmd = tonumber(ls[5])
    enf = tonumber(ls[6])
    
    ab = math.sqrt((xa - xb)*(xa - xb) + (ya - yb)*(ya - yb))
        
    cmd = (cmd * 3.14159265358979323846264338327950288419716939937510) / 180
    enf = (enf * 3.14159265358979323846264338327950288419716939937510) / 180
    
    print(string.format("%.3f", (ab * (1/math.tan(cmd) + 1/math.tan(enf)))))
end